packages/scene-composer/src/store/slices/SceneDocumentSlice.ts

```diff
@@ -45,7 +46,12 @@ export interface ISceneDocumentSlice {
   updateSceneRuleMapById(id: string, ruleMap: IRuleBasedMapInternal): void;
   removeSceneRuleMapById(id: string): void;
   addComponentInternal(nodeRef: string, component: ISceneComponentInternal): void;
-  updateComponentInternal(nodeRef: string, component: ISceneComponentInternal, replace?: boolean): void;
+  updateComponentInternal(
+    nodeRef: string,
+    component: ISceneComponentInternal,
+    replace?: boolean,
+    layerIdForEntityUpdate?: string,
```

Comment:
you don't need to pass this layerId in, this store has access to `getSceneProperty` and you can just get layerId with it. you also don't need to `useDynamicScene` because when we have layerId, we can assume it's enabled. so you don't need the changes above for the component editors. after my current change, the layerId will be stored in the node, so then we can just update here to get layerId from node instead of scene property.